src/linux/mod.rs

```diff
@@ -126,10 +128,18 @@ const BREAK: libc::c_int = 1;
 
 impl<'a> SharedLibrary<'a> {
     unsafe fn new(info: &'a libc::dl_phdr_info, size: usize) -> Self {
+        let mut name = Cow::Borrowed(CStr::from_ptr(info.dlpi_name));
+        if name.to_bytes().is_empty() {
+            name = if let Ok(exe) = current_exe() {
+                Cow::Owned(CString::from_vec_unchecked(exe.into_os_string().into_vec()))
+            } else {
+                Cow::Borrowed(CStr::from_bytes_with_nul_unchecked(b"/proc/self/exe\x00"))
```

Comment:
You can use \0 instead of \x00.

Reply:
Changed. Just personal preference on my part.